packages/mobile/src/screens/profile-screen/ProfileHeader/ProfileHeader.tsx

```diff
@@ -46,11 +47,66 @@ type ProfileHeaderProps = {
 export const ProfileHeader = (props: ProfileHeaderProps) => {
   const { scrollY } = props
   const styles = useStyles()
-  const profile = useSelectProfileRoot(['user_id', 'does_current_user_follow'])
+  const profile = useSelectProfileRoot([
+    'user_id',
+    'does_current_user_follow',
+    'current_user_followee_follow_count'
+  ])
   const accountId = useSelectorWeb(getUserId)
+  const { tier = 'none' } = useSelectTierInfo(profile?.user_id ?? 0)
+  const hasTier = tier !== 'none'
   const isOwner = profile?.user_id === accountId
+  const hasMutuals =
+    !isOwner && (profile?.current_user_followee_follow_count ?? 0) > 0
   const [hasUserFollowed, setHasUserFollowed] = useToggle(false)
+  const {
+    website,
+    donation,
+    twitter_handle: twitterHandle,
+    instagram_handle: instagramHandle,
+    tiktok_handle: tikTokHandle,
+    supporting_count: supportingCount
+  } = useSelectProfile([
+    'website',
+    'donation',
+    'twitter_handle',
+    'instagram_handle',
+    'tiktok_handle',
+    'supporting_count'
+  ])
+  const hasMultipleSocials =
+    [website, donation, twitterHandle, instagramHandle, tikTokHandle].filter(
+      Boolean
+    ).length > 1
+  const isSupporting = supportingCount > 0
   const [isExpanded, setIsExpanded] = useToggle(false)
+  const [isExpansible, setIsExpansible] = useState(false)
+
+  /**
```

Comment:
I like the comment, but wondering if the code is clear enough? like hasTier || hasMutuals || has.... seems quite nice to me!

Reply:
maybe create a "shouldExpand" const before the useEffect, and just pass that into dep array and check in if statement?

Reply:
one last thing, definitely not something we need to do, but could create an `<ExpandableHeaderDetails />` component that handles all the logic written here, which will nicely encapsulate this complex logic away from the root component, which can just focus on getting the core components rendering. its also a slight perf boost since rerenders related to that section don't trigger full rerender on the whole header

Reply:
yeah considered the shouldExpand var and went back and forth with myself. but yeah overall agree with the abovesss and will address